cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/TransactionPage.java

```diff
@@ -0,0 +1,281 @@
+    }
+
+    public String verifyTransactionDetails() {
+        return waitForElement(transactionsTableFirstRow, 10).getText();
+    }
+
+    // Read Transaction Methods
+    public Boolean printTransactionTable() {
+        System.out.println(waitForElement(transactionsTable, 10).getText());
+        return transactionsTable.isDisplayed();
+    }
+
+    // Update Transaction Methods
+
+    public void updateName(String name) {
+        waitForElement(getWebElement(editTransactionNameField), 10).clear();
+        getWebElement(editTransactionNameField).sendKeys(name);
+    }
+
+    public void updateAccount(String account) {
+        if (account != null && !account.isEmpty()) {
+            clickButton(account);
+            WebElement option = driver.findElement(By.xpath("//*[@id='edit-transaction-account']/option[text()='" + account + "']"));
+            option.click();
+        }
+    }
+
+    public void updateAmount(String amount) {
+        waitForElement(getWebElement(editTransactionAmountFieldId), 10).clear();
+        getWebElement(editTransactionAmountFieldId).sendKeys(amount);
+    }
+
+    public void updateCategory(String category) {
+        if (category != null && !category.isEmpty()) {
+            clickButton(category);
+            WebElement option = driver.findElement(By.xpath("//*[@id='edit-transaction-category']/option[text()='" + category + "']"));
+            option.click();
+        }
+    }
+
+    // Delete Transaction Methods
+    public void clickDeleteBtn() {
+        tableSizeBeforeDeletion = transactionsTable.getText().length();
+        waitForElement(deleteBtn, 10).click();
+    }
+
+    public Boolean confirmDeletion() {
+        int tableSizeAfterDeletion = transactionsTable.getText().length();
+        return tableSizeBeforeDeletion > tableSizeAfterDeletion;
+    }
+
+    // Category Methods
+
+    public void selectACategory(String category) {
+        if (category != null && !category.isEmpty()) {
+            clickButton(category);
+            WebElement option = driver.findElement(By.xpath("//*[@id='allCategoriesDropDown']/option[text()='" + category + "']"));
+            option.click();
+        }
+    }
+    public void printOutCategoryColumn() {
+        List<WebElement> categoryColumnValues = driver.findElements(By.xpath("//*[@id='root']/div[1]/main/div/div[3]/div/table//tr/td[3]"));
+        for (WebElement cell : categoryColumnValues) {
+            System.out.println(cell.getText());
+        }
+    }
+
+    @Override
+    public List<Component> getChildComponents() {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Unimplemented method 'getChildComponents'");
+    }
+
+    @Override
+    public Component getChildComponent(String name) {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Unimplemented method 'getChildComponent'");
+    }
+
+    @Override
+    public List<WebElement> getWebElements() {
```

Comment:
Medium impact: Use of getWebElements would throw an exception here. Should the need arise for use of the TransactionPage, this would make automation impossible.
Suggestion: return list of all web elements present in TransactionPage.